From bdd759af1113ba4a97edba5a58a864ccb0420aee Mon Sep 17 00:00:00 2001
From: "JianFeng,Zhou" <jianfeng.zhou@intel.com>
Date: Tue, 13 Oct 2020 13:32:34 +0800
Subject: [PATCH] [PATCH] Support SCSI storage for HyperV

HyperV only support SCSI storage. To run on HyperV, SCSI storage must be supported.

Tracked-On: OAM-92157
Signed-off-by: Xihua Chen <xihua.chen@intel.com>
Signed-off-by: JianFeng,Zhou <jianfeng.zhou@intel.com>
---
 init/devices.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/init/devices.cpp b/init/devices.cpp
index 9fbec64..6366264 100644
--- a/init/devices.cpp
+++ b/init/devices.cpp
@@ -340,6 +340,8 @@ std::vector<std::string> DeviceHandler::GetBlockDeviceSymlinks(const Uevent& uev
             symlinks.emplace_back("/dev/block/mapper/by-uuid/" + uuid);
         }
         return symlinks;
+	} else if (StartsWith(uevent.path, "/devices/virtual/block") || StartsWith(uevent.path, "/devices/LNXSYSTM")) {
+		// fall through to support other storage such as SCSI storage
     } else {
         return {};
     }
-- 
2.7.4

